packages/utilities/src/lib/utilityTypes.ts

```diff
@@ -72,3 +72,91 @@ export type Awaited<T> = PromiseLike<T> | T;
  * Type union for the full 2 billion dollar mistake in the JavaScript ecosystem
  */
 export type Nullish = null | undefined;
+
+/**
+ * Similar to the built in {@link NonNullable}, but properly removes `null` from all keys in the class or interface
+ * This does not recurse deeply, for that use {@link DeepRequired}
+ */
+export type NonNullableClass<T = unknown> = {
+	[P in keyof T]: NonNullable<T[P]>;
+};
+
+/**
+ * An object that is non nullable, to bypass TypeScript not easily working with {@link Record}<{@link PropertyKey}, unknown> in various instances.
+ */
+export interface NonNullObject {}
+
+/**
+ * Any object at all. This can be used if you need to assign the most generic type of object to a type.
+ */
+export type AnyObject = Record<PropertyKey, unknown> | NonNullObject;
```

Comment:
This type is useless, NonNullObject already includes the record as it's the base.

Reply:
Then why is it in Skyra